lib/go_script/go.rb

```diff
@@ -90,10 +90,11 @@ def build_jekyll(extra_args = '')
   end
 
   def git_sync_and_deploy(commands, branch: nil)
-    exec_cmd 'git stash'
-    exec_cmd "git checkout -b #{branch}" unless branch.nil?
-    exec_cmd 'git pull'
-    exec_cmd 'bundle install' if File.exist? 'Gemfile'
+    exec_cmd 'git fetch origin #{branch}'
+    exec_cmd 'git clean -f'
+    exec_cmd "git reset --hard #{branch.nil? ? 'HEAD' : 'origin/' + branch}"
+    exec_cmd 'git submodule --sync --recursive'
+    exec_cmd 'git submodule update --init --recursive'
```

Comment:
Yes, though note that the submodules will get synchronized to the versions currently checked into the branch, rather than synchronized to the latest versions available (which would be `git submodule update --remote`).

The nice properties of these commands are that they are no-ops when there aren't any submodules present at all.